access/src/main/java/zone/cogni/asquare/edit/delta/Delta.java

```diff
@@ -39,28 +42,30 @@ public Model getModel() {
   }
 
   public String getSparql() {
-    return Stream.of(getDeleteSparql(), getInsertSparql())
-            .filter(Objects::nonNull)
-            .collect(Collectors.joining(";\n"));
+    return Stream.concat(getDeleteSparql(), getInsertSparql())
+                 .collect(Collectors.joining(";\n"));
   }
 
-  private String getDeleteSparql() {
+  private Stream<String> getDeleteSparql() {
     List<Statement> removeStatements = getRemoveStatements();
-    if (removeStatements.isEmpty()) return null;
+    if (removeStatements.isEmpty()) return Stream.empty();
 
-    return "DELETE DATA { \n" +
-           convertStatementsToSparql(removeStatements) + "\n" +
-           "} \n";
+    return Lists.partition(removeStatements, PARTITION_SIZE)
+                .stream()
+                .map(statements -> "DELETE DATA { \n" +
+                                   convertStatementsToSparql(statements) + "\n" +
+                                   "} \n");
```

Comment:
I would put the ; here, then you don't have to "tweak" the query after you return it